### PR TITLE
Premis Mocking in IngestContentJobTest

### PR DESCRIPTION
**Pull request by bbpennel**

### CODE REVIEW

File: deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobTest.java [hunk 1]
```diff
@@ -79,10 +85,21 @@
 	private FolderObject destinationObj;
 
 	@Mock
+	private PremisLoggerFactory premisLoggerFactory;
```
Comment (harringj) [inline deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobTest.java]: What do you think about prepending these two variable names with "mock", just for consistency?

File: metadata/pom.xml [hunk 2]
```diff
@@ -19,6 +19,13 @@
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-jar-plugin</artifactId>
 				<version>2.3.2</version>
+				<executions>
```
Comment (harringj) [inline metadata/pom.xml]: What does this change do?

File: metadata/src/test/java/edu/unc/lib/dl/test/SelfReturningAnswer.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import static org.mockito.Mockito.RETURNS_DEFAULTS;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Mockito answer which returns the mocked object for all unstubbed method calls.
+ *
+ */
```
Comment (harringj) [inline metadata/src/test/java/edu/unc/lib/dl/test/SelfReturningAnswer.java]: This is AWESOME!!! Why have we not been using this all along?! :-)

Comment (bbpennel) [inline metadata/src/test/java/edu/unc/lib/dl/test/SelfReturningAnswer.java]: Because we hadn't done the right google searches yet :)

**Pull request merged**